tedge/src/mqtt.rs

```diff
@@ -91,7 +98,11 @@ impl Command for MqttCmd {
                 message,
                 qos,
             } => publish(topic, message, *qos)?,
-            MqttCmd::Sub { topic, qos } => subscribe(topic, *qos)?,
+            MqttCmd::Sub {
+                topic,
+                qos,
+                no_topic,
+            } => subscribe(topic, *qos, no_topic)?,
```

Comment:
Here `no_topic` is reference because of `&self` argument in this function.
I would add de-ref here like `*qos`. 
```suggestion
            } => subscribe(topic, *qos, *no_topic)?,
```

Then, in the function `subscribe()`, remove `&` from `no_topic`.
```
async fn subscribe(topic: &str, qos: QoS, no_topic: bool) -> Result<(), MqttError> {
```

Also, in the function `handle_messages()`, remove `&` from argument and remove `*` from if-statement.
```
async fn handle_message(message: Message, no_topic: bool) -> Result<(), MqttError> {
    if no_topic {
```

This looks better than now I think.